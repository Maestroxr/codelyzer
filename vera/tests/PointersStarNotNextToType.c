
int main()
{

	int x = 0;
	
	/*OK*/
	int* p1 = NULL;
	int* p2 = (int*)malloc(sizeof(int)),* p3 = NULL;
	int** p4 = (int**)malloc(sizeof(int)),***** p5 = (int*****)malloc(sizeof(int));
	
	/*--------*/
	
	/*PROBLEM*/
	int * problem1 = NULL;
	int* pp2 = (int*)malloc(sizeof(int)), * problem2 = NULL;
	int*  * p4 = (int**)malloc(sizeof(int)),  ** *  **     * p5 = (int******)malloc(sizeof(int));
	
	/*********/
	return x;
}



