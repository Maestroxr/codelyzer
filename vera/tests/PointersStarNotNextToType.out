C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNotNextToType.c:9: Malloc lacks a pointer cast.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNotNextToType.c:9: Pointer's 'p2' Malloc(s) missing '1' null pointer check(s).
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNotNextToType.c:10: Malloc's cast is to incorrect type, pointer initialized as 'int**********' but cast was to 'int*****'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNotNextToType.c:15: Pointer's '*' not next to the pointer's type 'int* problem1' instead of 'int * problem1'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNotNextToType.c:16: Pointer's '*' not next to the pointer's type 'int ,* problem2' instead of 'int , * problem2'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNotNextToType.c:16: Malloc lacks a pointer cast.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNotNextToType.c:16: Pointer's 'pp2' Malloc(s) missing '1' null pointer check(s).
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNotNextToType.c:17: Malloc's cast is to incorrect type, pointer initialized as 'int************' but cast was to 'int******'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNotNextToType.c:17: Pointer's '*' not next to the pointer's type 'int****** p4' instead of 'int*  ***** p4'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersStarNotNextToType.c:17: Pointer's 'p4' Malloc(s) missing '4' null pointer check(s).
